src/renderer/dx/DxRenderer.cpp

```diff
@@ -285,12 +285,24 @@ HRESULT DxEngine::_SetupTerminalEffects()
 
     RETURN_IF_FAILED(_d3dDevice->CreateBuffer(&bd, &InitData, &_screenQuadVertexBuffer));
 
+    D3D11_BUFFER_DESC pixelShaderSettingsBufferDesc{};
+    pixelShaderSettingsBufferDesc.Usage = D3D11_USAGE_DEFAULT;
+    pixelShaderSettingsBufferDesc.ByteWidth = sizeof(_pixelShaderSettings) + (16 - sizeof(_pixelShaderSettings) % 16) % 16;

```

Comment:
Okay I don't know DX that well but _what the heck is going on here_

Reply:
I believe this is "nearest larger multiple of 16"

```
libra  ~  %   $n=9
libra  ~  %   $n + (16 - $n % 16) % 16
16
libra  ~  %   $n=17
libra  ~  %   $n + (16 - $n % 16) % 16
32
```

Reply:
(that "multiples of 16" thing is documented in the docs for `CreateBuffer` for when you pass `D3D11_BIND_CONSTANT_BUFFER`)

Reply:
Right. Thinking about this more, I want to pad the struct instead. As it is now, I would assume DirectX is reading beyond the struct's memory when initializing / updating the constant buffer, which would make me feel bad.

Reply:
Thanks for pointing that out, @zadjii-msft !